src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -1210,6 +1219,18 @@ public function count(): int
         return \count($this->defaults);
     }
 
+    /**
+     * Sets ignore undefined options.
+     *
+     * @return $this
+     */
+    public function ignoreUndefined(): static
```

Comment:
To maintain consistency in the design of this component, this method (as is) should be relocated to the fluent interface within `OptionConfigurator`. And here, it should be transformed into a traditional setter: `setIgnoreUndefined(bool $value): static`.

Reply:
I didn't get the thing about moving method to the OptionConfigurator.
This method sets the behaviour of whole OptionResolver instance not particular option inside of it.
please explain me the idea of it to be able do what you ask

Reply:
The goal of the `OptionConfigurator` is to be able to write this code:
```php
        $resolver
            ->define('a')
                ->required()
                ->allowedTypes('string')
            ->define('b')
                ->default(10)
                ->allowedTypes('int')
            ->ignoreUndefined();
```

Currently, the call the `ignoreUndefined()` have to be first.

```php
        $resolver
            ->ignoreUndefined()
            ->define('a')
                ->required()
                ->allowedTypes('string')
            ->define('b')
                ->default(10)
                ->allowedTypes('int');
```

But @yceruto I would no name the method differently in `OptionResolver` and `OptionConfigurator`, so that the order of method calls does not matter.

Also, I've not found if it was discussed, the method could accept an argument if someone needs to reset the behavior: `function ignoreUndefined(bool $ignoreUndefined = true)` should accept a boolean if someone

Reply:
thanks for the advise.
i did changes

Reply:
I'm mostly okay with the method name, and I personally prefer the `ignoreUndefined` version, but I'm more worried about having two different ways to name boolean methods in the same class. Right now, we have `setPrototype(bool $prototype)` in `OptionResolver` class, so I will go for consistency here.

Yes, this new method isn't related to a specific option, but it can help make the `OptionConfigurator` API easier to use as explained by @GromNaN.

Reply:
so what here next?